src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -197,40 +256,69 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
         return writer.EndContainer(fabricSceneContainer);
     }
 
+    /// @brief This Deserialize method needs the member scene_storage to be defined previously (in this case in the load function).
+    /// It checks that the recovered scenes from the deserialization fit in the current max and if there are too many scenes in nvm,
+    /// it deletes them. The method sets the deleted_scene member to tru if scenes were deleted so that the load function can know
+    /// it needs to save the Fabric scene data to update the scene_count and the scene map in stored memory.
+    /// @param reader TLV reader, must be big enough to hold the scene map size
+    /// @return CHIP_NO_ERROR on success, specific CHIP_ERROR otherwise
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override
```

Comment:
I guess the point is that we can't change the signature here?

But... who is actually calling Deserialize?  Before this PR, it's the virtual hook used by Load, with Load non-virtual.  But we're not virtualizing Load and implementing it ourselves.  So the only caller of Deserialize is our Load override, which we control.

So we could change Deserialize to just return error, have another method on this class that does the deserialization work, is called from Load, and takes whatever arguments we want instead of passing things in/out through members), right?

Reply:
We could indeed use another method than Deserialize, however we still need to implement Deserialize as it is purely virtual. The persistant Data class is used in all the class getting stored and also in the group data provider. 

My alternative solution would be to create a separate function and implement Deserialize as {return CHIP_NO_ERROR;}

Reply:
> My alternative solution would be to create a separate function and implement Deserialize as {return CHIP_NO_ERROR;}

That's what I suggested above, no?

Reply:
Indeed, Friday afternoon me must have understood things differently somehow...

Reply:
Last commit does as suggested.